hack/ci-e2e-kind-migration.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
```

Comment:
For symmetry reasons, should we rename this file to `test-e2e-migration-local.sh`?

Reply:
I proposed `hack/ci-e2e-kind-migration.sh` for symmetry with the to-be-introduced `hack/ci-e2e-kind.sh`, ref https://github.com/gardener/gardener/pull/6089#issuecomment-1164029426

Reply:
OK, I see. Should we use the opportunity to introduce this `ci-e2e-kind.sh` now? 👀